### PR TITLE
feat: added modal confirmation before pinning message

### PR DESCRIPTION
**Pull request by Sayan4444**
Proposed changes (including videos or screenshots)

AFTER THE FIX
https://github.com/RocketChat/Rocket.Chat/assets/112304873/2e74a44f-e168-48af-817d-c7b945d1f5b0

REASON BEHIND CHANGES
Made the same changes as done by DISCORD to tackle this issue.

Issue(s)
This should close issues :-
closes https://github.com/RocketChat/Rocket.Chat/issues/30777

Steps to test or reproduce

1. Go to Teams/Channel
2. Click on any of the teams/Channels where you are participant.
3. Try to Send a Text to the Team/Channel
4. Try to Edit the Text and See that there is no Cancel Button for Edit Text if the user wants to continue without editing the Text.]

### CODE REVIEW

File: apps/meteor/client/startup/actionButtons/pinMessage.ts [hunk 1]
```diff
@@ -18,13 +20,31 @@ Meteor.startup(() => {
...
+				} catch (error) {
+					dispatchToastMessage({ type: 'error', message: error });
+				}
+				onCloseModal();
+			};
+
+			imperativeModal.open({
+				component: GenericModal,
+				props: {
+					title: 'Are you sure you want to pin this message?',
```
Comment (dougfabris) [inline apps/meteor/client/startup/actionButtons/pinMessage.ts:38]: avoid using raw texts, let's use translations instead

File: apps/meteor/client/startup/actionButtons/pinMessage.ts
Comment (dougfabris) [inline apps/meteor/client/startup/actionButtons/pinMessage.ts:1]: Let's add the message that will be pinned in the Modal body

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/client/startup/actionButtons/pinMessage.ts
Comment (hugocostadev) [reply apps/meteor/client/startup/actionButtons/pinMessage.ts:1]: @Sayan4444 you can check the Forward message modal to check how you can accomplish that

Comment (Sayan4444) [reply apps/meteor/client/startup/actionButtons/pinMessage.ts:1]: Thanks a lot I was also looking for an example

**Review by dougfabris — CHANGES_REQUESTED** [review]
@Sayan4444 This is the desired result shared by our design team, please check if you can achieve it using `<GenericModal />`, if you have any questions, let me know!



Also, can you check what happened with the `en.i18n.json`?

**Comment by Sayan4444** [comment]
@dougfabris Is this good enough or do you want me to add or correct anything esle??

Screenshot 2024-01-03 224853

Can you tell me what is wrong about en.i18n.json, it will be of greate help

**Review by dougfabris — APPROVED** [review]
@Sayan4444 Thank you so much for the contribution, it was very precise and valuable. 
About the `en.json` file I think it was some issue syncing your code or with lint.
I made some small changes, just to match our standards and code organization.
Thanks again and I hope to see you contributing more 🚀

**Review by dougfabris — DISMISSED** [review]

**Comment by Sayan4444** [comment]
@dougfabris Yep I will be contributing more as much as I can as I love solving issues which directly impact the end user experience. Do you have an issue which u are also waiting long time to see someone working on it?

**Comment by dougfabris** [comment]
@Sayan4444 That's awesome to hear from you. 
Feel free to navigate through our repository and have fun with whatever you want, we found your choices amazing and we'll be reviewing them as soon as possible!

**Review by hugocostadev — APPROVED** [review]

**Pull request merged**